### ISSUE TITLE
How to set BPPARAM to run parallel

### ISSUE THREAD

**Issue by lindsaynhayes** [HelenaLC/muscat#70]
How do you specify the `BPPARAM` argument to make `mmDS` to run parallel? The default is set as `BPPARAM = SerialParam(progressbar = verbose)` which means it going to run serial right? And when I run it I see the iterations ticking by (<3 verbose). But for one coef it is going to take over 7 hours so I was hoping to speed it up. I've been reading the BPPARAM vignette , but I'm still confused.  Below is what I've tried. 
(http://bioconductor.org/packages/release/bioc/vignettes/BiocParallel/inst/doc/Introduction_To_BiocParallel.pdf)
```
library('BiocParallel')
registered() # I have MulticoreParam capabilities
options(MulticoreParam=quote(MulticoreParam(workers=12))) # from page 3
param <- MulticoreParam(workers = 12) # from page 5

res_MC <- mmDS(sce, coef = "group_idCoef2", method = "dream", verbose = TRUE, BPPARAM = MulticoreParam())

res_MC <- mmDS(sce, coef = "group_idCoef2", method = "dream", verbose = TRUE, BPPARAM = MulticoreParam(workers = 8))

res_MC <- mmDS(sce, coef = "group_idCoef2", method = "dream", verbose = TRUE, BPPARAM = param)

# they all give an error with some version of 
 unused argument (BPPARAM = MulticoreParam())
```

Thanks for any advice. I'm still trying to get a handle on parallel processing.

**Comment by plger** [comment]
There seems to be nothing wrong with what you're doing, but I can't reproduce the bug neither in version 1.6.0 (Bioc) or 1.7.1 (Bioc-devel). Are you perhaps using an older version?

**Comment by lindsaynhayes** [comment]
You are correct. I was assuming I had the syntax wrong. I was running muscat_1.4.0. Does version 1.6 or 1.7 run on R 4.0 because that is the newest version on the HPC? When I tried to install I get `ERROR: this R is version 4.0.2, package 'muscat' requires R >= 4.1`
```
> sessionInfo()
R version 4.0.2 (2020-06-22)
Platform: x86_64-conda_cos6-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)
 
other attached packages:
[1] muscat_1.4.0
```

**Comment by plger** [comment]
On what OS are you running this? Depending on the platform BiocParallel sometimes has issues, especially when running it interactively from something like Rstudio, so you might try from the CLI. Otherwise I'm afraid `dream` has a big memory consumption, though I don't see why your setup wouldn't be enough...

**Comment by lindsaynhayes** [comment]
1. I am using CLI from a HPC to (try to) run Parallel. Here is the abbreviated sessionInfo. ( I can send more, but didn't think you really needed all the package info) 
```
> sessionInfo()
R version 4.1.0 (2021-05-18)
Platform: x86_64-conda-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

[13] muscat_1.6.0  
```
2. The serial ended up being faster. I played around with 4, 6, 8 workers and the speed was pretty much the same for each. Though with `log=TRUE` I could see the different cores running each iteration. But I ended up getting `dream` and `vst` with `sctransform` to work in serial and stopped trying with the parallel. 

3. `dream2` was running slower than `dream` so I didn't wait for that one to finish. For `dream2` each cluster was running 2 rounds of 100 iterations, is that normal? While `dream` was just running one set of 100 iterations for each cluster.
